src/goto-symex/symex_assign.cpp

```diff
@@ -36,6 +36,14 @@ void goto_symext::symex_assign_rec(
   code_assignt deref_code=code;
 
   clean_expr(deref_code.lhs(), state, true);
+  // make the structure of the lhs as simple as possible to avoid,
```

Comment:
Apologies for the delay in responding. It refers to the function `do_simplify`. I believe this comment to be explaining this function (all the way down to line 45).